diagnostics/include/diagnostics_server.h

```diff
@@ -69,11 +69,11 @@ inline std::string readline(int sock) {
       throw std::runtime_error("Request exceeded max size: " + std::to_string(MAX_INPUT_SIZE));
     }
 
-    ssize_t read_rv = read(sock, buf.data() + count, buf.size() - count);
+    const auto read_rv = read(sock, buf.data() + count, buf.size() - count);
```

Comment:
I specifically put `ssize_t` there, because auto creates an `int` which is a narrowing conversion in some cases.

Reply:
Do you refer to `read_rv` being `int`? It should be the return type of read(), i.e. `ssize_t`. I suspect I didn't get what you meant, though. Please correct me if I am wrong.

Reply:
You are correct. Sorry, I was thinking of something else.